### ISSUE TITLE
Datahub module not found

### ISSUE THREAD

**Issue by willjharmer** [datahuborg/datahub#119]
When building and running with vagrant, I consistently get a successful vagrant up (and provision) but when accessing the site I then get a django error page, telling me  'No module named datahub'. I then go into the app container and have to rerun the thrift generators for datahub.thrift. Then everything works.

Error page and environment is attached.
[No module named datahub.pdf](https://github.com/datahuborg/datahub/files/161222/No.module.named.datahub.pdf)

**Comment by RogerTangos** [comment]
This is an ongoing problem. Unfortunately, it's very hard for us to debug, since it happens during vagrant setup, and doesn't seem to happen consistently.

@willjharmer  - I know you've already fixed this, but in case anyone else encounters it, here are some commands that should fix the problem:
```
# log into vagrant and become the superuser
$ vagrant ssh
$ sudo su

# go into the app container
$ docker exec -ti app /bin/bash

# run the setup script that was supposed to run at start
$ source src/setup.sh

# exit the app container
$ exit

# restart containers
$ dh-start-containers
```

If this doesn't work for you, please ping us on this thread and @justinanderson or I will give you a hand. This is a known and annoying issue, that's been particularly difficult to track down.

**Comment by RogerTangos** [comment]
resolved in d02eb039c52a96a2071e89a4071240e65d22d991

**Comment by justinanderson** [comment]
This should be fixed by https://github.com/datahuborg/datahub/commit/f552f2ba75e733aee856f53f10edb47b530a6076, which was just merged into master and tagged as v0.4 yesterday. You can upgrade with:

```bash
git checkout master
git pull
sudo su
# Rebuild the Docker images
dh-build-images
# Stop the containers
dh-stop-containers
# Delete the app and web containers
docker rm app web
# Recreate the app and web containers (ignoring complaints that the other containers already exist)
dh-create-dev-containers
# Restart DataHub
dh-start-containers
# Once migrations finish and the script returns control, DataHub should be up and running again.
```
